app/models/grda_warehouse/tasks/identify_duplicates_query_matcher.rb

```diff
@@ -0,0 +1,241 @@
+          destination_data_source_ids: destination_data_source_ids,
+          unprocessed_ids: unprocessed_ids,
+        )
+      end
+
+      # Builds a query matcher for DOB matches
+      # @param match_type [Symbol] :existing or :unprocessed
+      # @param destination_data_source_ids [Array<Integer>] Required for :unprocessed match_type
+      # @param unprocessed_ids [Array<Integer>] Required for :unprocessed match_type
+      # @return [IdentifyDuplicatesQueryMatcher] Query matcher instance (call .execute to run)
+      def self.for_dob_matches(match_type:, destination_data_source_ids: nil, unprocessed_ids: nil)
+        ensure_match_type!(match_type)
+
+        new(
+          match_type: match_type,
+          field_expression: %(clients."DOB"),
+          field_alias: 'dob',
+          field_filters: DOB_FILTERS,
+          include_value_in_results: false,
+          destination_data_source_ids: destination_data_source_ids,
+          unprocessed_ids: unprocessed_ids,
+        )
+      end
+
+      def self.ensure_match_type!(match_type)
+        return if MATCH_TYPES.include?(match_type)
+
+        raise ArgumentError, "Unsupported match_type: #{match_type.inspect}"
+      end
+      private_class_method :ensure_match_type!
+
+      def initialize(match_type:, field_expression:, field_alias:, field_filters:, include_value_in_results:, destination_data_source_ids: nil, unprocessed_ids: nil)
+        @match_type = match_type
+        @field_expression = field_expression
+        @field_alias = field_alias
+        @field_filters = field_filters
+        @include_value_in_results = include_value_in_results
+        @destination_data_source_ids = sanitize_ids(destination_data_source_ids)
+        @unprocessed_ids = sanitize_ids(unprocessed_ids)
+      end
+
+      def to_sql(warehouse_id: nil)
+        @warehouse_id = warehouse_id || GrdaWarehouse::DataSource.warehouse_id
+        match_type == :existing ? existing_sql : unprocessed_sql
+      end
+
+      # Executes the query and returns processed ID pairs
+      # @param warehouse_id [Integer] Warehouse data source ID (defaults to system warehouse)
+      # @param validate_ssn [Boolean] Whether to validate SSNs using HudHelper (default: true for SSN matches)
+      # @return [Array<Array<Integer>>] Array of ID pairs, or empty array if no unprocessed_ids
+      def execute(warehouse_id: nil, validate_ssn: nil)
+        # Default validate_ssn to true for SSN matches
+        validate_ssn = (field_alias == 'ssn') if validate_ssn.nil?
+
+        # Early return for unprocessed with no IDs
+        return [] if match_type == :unprocessed && unprocessed_ids.blank?
+
+        results = GrdaWarehouse::Hud::Client.connection.execute(to_sql(warehouse_id: warehouse_id))
+
+        # Filter SSNs if validation requested and SSN is in results
+        results = results.select { |r| ::HudHelper.util.valid_social?(r['ssn']) } if validate_ssn && include_value_in_results && field_alias == 'ssn'
```

Comment:
isn't `&& field_alias == 'ssn'` duplicative?  Potentially someone could pass `validate_ssn: true` on some other field, but that seems unlikely.  That said, the check is not expensive.